.github/workflows/wheels.yml

```diff
@@ -130,16 +130,25 @@ jobs:
           CIBW_ENVIRONMENT_PASS_LINUX: RUNNER_OS
           CIBW_PRERELEASE_PYTHONS: True
 
+          # TODO remove the CIBW_BEFORE_BUILD_* CIBW_BEFORE_TEST lines once there are
+          #  numpy2.0 wheels available on PyPI.
+          # Also remove PIP_NO_BUILD_ISOLATION flag.
+          # For some reason asv needs extra packages for installation on cp312,
+          # but not on lower python versions
+          CIBW_BEFORE_BUILD_LINUX: "pip install numpy>=2.0.0.dev0 meson-python cython pythran pybind11 ninja; bash {project}/tools/wheels/cibw_before_build_linux.sh {project}"
+          CIBW_BEFORE_BUILD_WINDOWS: "pip install numpy>=2.0.0.dev0 meson-python cython pythran pybind11 ninja && bash {project}/tools/wheels/cibw_before_build_win.sh {project}"
+          CIBW_BEFORE_BUILD_MACOS: "pip install numpy>=2.0.0.dev0 meson-python cython pythran pybind11 ninja; bash {project}/tools/wheels/cibw_before_build_macos.sh {project}"
+          CIBW_BEFORE_TEST: pip install setuptools setuptools_scm[toml] wheel
           # required so that cp312 can grab a nightly wheel
           # can remove when cp312 is released and we don't need to search
           # other wheel locations.
-          # PIP_EXTRA_INDEX_URL=https://pypi.anaconda.org/scientific-python-nightly-wheels/simple
-          CIBW_ENVIRONMENT: PIP_PRE=1
+          CIBW_ENVIRONMENT: "PIP_NO_BUILD_ISOLATION=false PIP_PRE=1 PIP_EXTRA_INDEX_URL=https://pypi.anaconda.org/scientific-python-nightly-wheels/simple"
 
           CIBW_ENVIRONMENT_WINDOWS: >
             PKG_CONFIG_PATH=c:/opt/64/lib/pkgconfig
             PIP_PRE=1
-            # PIP_EXTRA_INDEX_URL=https://pypi.anaconda.org/scientific-python-nightly-wheels/simple
+            PIP_EXTRA_INDEX_URL=https://pypi.anaconda.org/scientific-python-nightly-wheels/simple
+            PIP_NO_BUILD_ISOLATION=false
```

Comment:
Obviously not a SciPy problem, but the naming of this variable/how it is activated is so bad I had to double take--looks like I'm not the only one: https://github.com/pypa/pip/issues/5735

You activate the true meaning of the variable by setting it to a false or null value, which is also compounded by parsing double negative logic when reading it the first time.

Reply:
Yes, it's pretty bad. 
Note for the future - cibuildwheel is currently unable to send flags/arguments to the build front end. Adding this functionality is being added [here](https://github.com/pypa/cibuildwheel/pull/1588)

Reply:
It'd be useful to put a comment about this _enabling_ build isolation.